tokio/src/runtime/mod.rs

```diff
@@ -183,7 +183,7 @@
 mod tests;
 
 cfg_rt_core! {
-    mod basic_scheduler;
+    pub(crate) mod basic_scheduler;
```

Comment:
I think this suggests that the `Queue` type should be moved out of `basic_scheduler` now that it's shared.

Reply:
I thought leaving it here kind of gave the implication that the `LocalSet` was "reusing" code from `basic_scheduler`, which seemed right to me? But I can move it if you'd prefer.

Reply:
Yeah, I can see that too. I just think that with this refactor arguably `Queues` is its own primitive that both `basic_scheduler` *and* `LocalSet` are using. Moving it out makes `Queues` more clearly "its own thing".

Reply:
hmm, I don't have a strong opinion on this, so I'd be fine with moving it out. What module would you put it in? The `Queues` type is specifically for a _single-threaded_ scheduler, so the module it goes in shouldn't have a name that implies the thread pool also uses this...

Reply:
`mod task_queue`? Is there a particular reason why we can't also share the queue impl that the thread pool uses? It also has a notion of local and remote tasks, right?

Reply:
> Is there a particular reason why we can't also share the queue impl that the thread pool uses? It also has a notion of local and remote tasks, right?

Yes, but it's significantly different. In order to implement work stealing, the threadpool workers need to allow other workers to steal tasks from their queues. This necessitates some complicated lock-free structures. In the local set and basic scheduler cases, there are fewer synchronization points. Carl might be able to better describe the differences, but this code is not really shared with the threadpool.

Reply:
Ah, yes, of course, you're right. How about `mpsc_task_queue`?

Reply:
I moved the queues to their own module (`task::queue`) in fa878c6aa5cf31770e1d2a4a6270f029774aa570, and renamed the struct to `SingleThreadQueues`, which I didn't really like...`mpsc` seems neater, thanks!

Reply:
:+1: I think this `pub(crate)` modifier can then be removed again, right?

Reply:
yup, good eye! i missed that!